src/analysis/retail/shaman/restoration/modules/dragonflight/Tier31.tsx

```diff
@@ -0,0 +1,151 @@
+import ITEMS from 'common/ITEMS';
+import talents from 'common/TALENTS/shaman';
+import { TIERS } from 'game/TIERS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { HealEvent } from 'parser/core/Events';
+import { isFrom4pcT31 } from '../../normalizers/CastLinkNormalizer';
+import { SHAMAN_T31_ID } from 'common/ITEMS/dragonflight';
+import { SpellLink } from 'interface';
+import ItemSetLink from 'interface/ItemSetLink';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import Statistic from 'parser/ui/Statistic';
+import { formatNumber, formatPercentage } from 'common/format';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import RiptideTracker from '../core/RiptideTracker';
+import Combatants from 'parser/shared/modules/Combatants';
+
+const RIPTIDE_4P_POWER = 0.2; // Riptide bonus power = 25% => Tier set value = 25%/125% = 20% of total amount
+//const debug = false;
+
+/**
+ * **Resto Shaman T31 (Amirdrassil)**
+ *
+ * 2pc: Chain Heal, Healing Surge, and Healing Wave mark their initial target with a Tidal Reservoir, causing them to receive 15% of all Riptide healing you deal for 15 sec.
+ *
+ * 4pc : Riptide's healing is increased by 25%. If Riptide is active on the same target as Tidal Reservoir, its heal over time effect has a 6% chance to create a new Riptide on a nearby ally.
+ */
+
+export default class Tier31 extends Analyzer {
+  static dependencies = {
+    riptideTracker: RiptideTracker,
+    combatants: Combatants,
+  };
+  protected riptideTracker!: RiptideTracker;
+  protected combatants!: Combatants;
+  has4pc: boolean;
+  tidalReservoirHealing: number = 0;
+  tidalReservoirOverhealing: number = 0;
+  bonus4pcRiptideHealing: number = 0;
+  percent4pcRiptideHealing: number = 0;
+  bonus4pcRiptideAmount: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.has2PieceByTier(TIERS.T31);
+    this.has4pc = this.selectedCombatant.has4PieceByTier(TIERS.T31);
```

Comment:
you don't need this field or to check it in your methods---they won't be called if `this.active` is false.

Reply:
Think this is fine since the stat has two separate line items for 2pc and 4pc - hides the 4pc block when only 2pc is equipped

Reply:
oh, you're right